### ISSUE TITLE
Update annotate-snippets to 0.11.0

### ISSUE THREAD

**Issue by MichaReiser** [astral-sh/ruff#10832]
See https://github.com/astral-sh/ruff/pull/10699

CC: @carljm in case you plan to work on this.

We may also decide not to use annotate-snippets anymore (it has some weirdness like it supports some markup that can mess up rendering)

**Comment by dhruvmanila** [comment]
This will be useful for the new parser test cases as well. If I understand this correctly,  https://github.com/rust-lang/annotate-snippets-rs/pull/107 allows highlighting empty spans which can occur at EOF. You might've noticed that there are no error messages in some of the cases :)

**Comment by MichaReiser** [comment]
Yeah. I think we should also change the expected errors to point to an empty range that starts at the next token's position instead of using the entire range of the next token.
